riscv/insns/fround_h.h

```diff
@@ -0,0 +1,18 @@
+require_extension(EXT_ZFH);
+require_extension(EXT_ZFA);
+require_fp;
+{
+  bool isNaN = isNaNF16UI(FRS1_H.v);
+  bool isZero = fracF16UI(FRS1_H.v) == 0 && expF16UI(FRS1_H.v) == 0;
+  if (isNaN || isZero) {
+    /* set invalid operation flag for signalling NaN */
+    if (f16_isSignalingNaN(FRS1_H))
+      STATE.fflags->write(STATE.fflags->read() | softfloat_flag_invalid);
+    /* NaN or positive/negative zero are copied unmodified */
+    WRITE_FRD_H(FRS1_H);
```

Comment:
NaNs must be canonicalized.

Reply:
Agreed, after reviewing our SAIL implementation.
The specification test does not explicitly state this, though.